### ISSUE TITLE
Help Updating the Axios libdef for flow v0.37.x

### ISSUE THREAD

**Issue by bassettsj** [flowtype/flow-typed#558]
I am really confused about the new error message that is coming up after flow was updated to 0.37.x . I would be happy to help contribute a fix, but I am really not certain how to correct the new messages that are showing up.

```
109:   declare interface AxiosExport extends Axios {               ^ property `Axios`. Property not found in
. Object
109:   declare interface AxiosExport extends Axios {               ^ property `CancelToken`. Property not found in
. Object
109:   declare interface AxiosExport extends Axios {               ^ property `Cancel`. Property not found in
. Object
109:   declare interface AxiosExport extends Axios {               ^ property `all`. Property not found in
. Object
109:   declare interface AxiosExport extends Axios {               ^ property `create`. Property not found in
. Object
109:   declare interface AxiosExport extends Axios {               ^ property `isCancel`. Property not found in
. Object
109:   declare interface AxiosExport extends Axios {               ^ property `spread`. Property not found in
. Object
```

[Example showing the error messages](https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVATApgYxgQwCcswBbODAVxhIHJ8APASzgGdawBvVMMbPIiSYA7AC5ZCUfDhIAFQnAYBPAMJxhUJgHMuPXmAAWbUQC4wrUYRFaA3Ht4AHOIVNhhlUgCMJd3gF89fgJiMBFxSWkSFXxhGRhdfTAcdQtCShxRZwAKUixWVnwtLAB+M1TrAEozaNisGF99XPzCrDLLawaA3iDBULEJKRkwGriJBP0cvIKi0vN24S0qsAA3OCYMTsDcYKF+iKGRuoAVOABrLGEAZThKQiHuRMzz4WqYuJPnht4cN7rX2pohE23W2vTCA0iw1+MA+F3G3xSlnSmUIWSwDFwlBRZiyPwB-1GhAqYAAvAA+FZrDBLQ4ws4XL5zfCiJg4cw3O5YLI06Gwq4cmSMhwKUhMVitMDyOCi8UAHlpZMZxHwrHUs1pjNEBgUCAAklAAEpYACOlDy4gw3LMq3WwL4oJC4P2JAAgsw2AANAASBrUGm0ACEVVhZUcKQ99PgMPgHOFZqGybj1JotGY3SxWN7fcntAnieTJSKxSH056fQnFfYwPgsQZZhHEpRxYRhPhcm0rAsADRVxwq1gIZwYDvWKt+RmeYMAVQNABlZuVu1W8e96cJ1by14yDFgoxJWLMAPKeABWuFE29EogcLqKYlmooxGzAwGAYAAsmLWNYwFqSKIlAcXZfx3JJnBIWBEDcCgSBgJhPGwKAqwMK8HFYW8LlEB8mCfGwXzfT98h-P9f0A4CSOSEJIKQYQYLAOCEKwJDElIRg-XEMRZwuLQtVmdwvB8KtWIYI0MCYYgMgPMwAFYe0SBwiDbKSwGPM8MiFRTSFYS4JCYfA4IALwkWYsgUwglLMVTz3zClF1sKthTgLRiHyEzHOcqYAFFlkwsxvMwmzKXWMAAB8yBwrANgchRlFmKVlD9FMlTyJxhHFI4yNmehCHMpRPEoKAoAkDgwtoTx4E8EqwFoDA4BwDxMKq2gT1VYQmvEBhRCa1Jd1IWhNSYXIbiwsx+O8IEq0sGJWCgZxSCNU1zVmF0cvwJRZVlKdE2jUR8DMI5AqnEKVtyjatqyHa9rAA7SQpLbK0ecy0tmwh5pSlISjTVb1s27bmSum6Cy2xlln09ZmSwS5dtEJsTIsZkm1GjxxsCzw4DgGgYjk-QECYLUVGIbAxD0mBlLRjHd2ERkGFYSQ1DOJgsAAOTbT65k7ezEhpyQvV3bBCBZ3IF3mTmwC6e0BBCACgLAAASUtMx9RLA2DBNSTABWs2VrQgzlMMGh6R09kGV13UV7N-S0NX0Q4jBWA1s2tZzHXVbDeE+H+2YjkZXItQoYWOcZW4YBHBY7UN-8yLlzWledtWSQdjMncthMDYdEgBHyROywNNWGzAy200d2OU-1qtLv27c+f3SzT3PRl4Zh1gkYEibEkbpsjht0OtGx3hiEWiwzBQ69ZZUODMIWs0LFCsAPXfWcvVQqfzT0cWI9ImX5eL3O3YT7ek-LMuQUljOCCzhXdT2GRY2cXUidEMAbYuO2wChjmuHX9OknP+2FZXiwV9wg3xRHnKsTYuS9nMOkGQ+QvQxAwICMwxQsguVSuKIuh8LYpjzLdQs0piyyhjtg3MIAyQUjCsQ7WsoyF930BIBQhB4HCEQRIZBaIcrODMI+SKgUeEYDoUsS+18sC30IPfTCjIsBqVEFkdYvls6sCARIEBd8H5LBtFFfwWxT4-37Ioo0rB0FYGUZyMRYDGziiyFA1gMCpjMNYYQdhaCPqYJzrggs-C6G8AYc4BxSCwAoN8U48KT4+ERQEVWIRZtTGqPEQ-KRMi5EPzcUokRYiJFiA0VScO39pamwzFKGUIY974OKUQneFY066Mzn-M27tkhpSRBkZwCYkyW2WjvbWusSlkmybaKsbTGkplScnHBYZomFKLHrB6+g2nBx7l2AuKZOlYOoRM1JRTCHH30APaeoghnO1GSXcZfTNnTN6YybANBxDzMICHdm1glnDO0KsnO3TXZnMUVsmZjIigHLDFkBZjzuzLNecckhLsZmTLYD8y5yE+Z3IeXZZ5zs3nmw+dC85BDfkOWMEixZHtdrYSfKijpELMW9JhawOFqcHJYgJSC3uRL8AksimSlZFLnY9LzNi8pOy+yiBwAYRlKKWVsoEWCrQ6Kxkqyxd8i5dLEhOjic3d2-cTT7NSQA0QsTRGgP4WSbxYAXFpQlP-d6Zq9XmMNXQ8ca90Anx2Hoi+ZtPKcMINbTqL97busYQ0o5ijZVWwFSay1GCg1H1mQibAAcOgOp0c6-J0c3Uevjoov1rSy6JrBMbSECtPIMCcC4J+3qWF1IzO7BWZh8lwCgIoxktJqi-3lNCaNUIAR8mbf2Vtna1ztrFLSLIoMYBmjMDEJQSxyaYypsuZU4h2mcsjZCnlE6vkK0ZPpB5tb61woAHRbobsKPmuIt2ThwKcMwAAxSgsQWTqCWFkIgISTprVlGuwKN670sGEA6p1vRyBUBoHu9ExbRBqoLUW5wF5UABCAA)

**Comment by marudor** [comment]
I know that this fixes [tryflow](https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVATApgYxgQwCcswBbODAVxhIHJ8APASzgGdawBvVMMbPIiSYA7AC5ZCUfDhIAFQnAYBPAMJxhUJgHMuPXmAAWbUQC4wrUYRFaA3Ht4AHOIVNhhlUgCMJd3gF89fgJiMBFxSWkSFXxhGRhdfTAcdQtCShxRZwAKUixWVnwtLAB+M1TrAEozaNisGF99XPzCrDLLawaA3iDBULEJKRkwGriJBP0cvIKi0vN24S0qsAA3OCYMTsDcYKF+iKGRuoAVOABrLGEAZThKQiHuRMzz4WqYuJPnht4cN7rX2pohE23W2vTCA0iw1+MA+F3G3xSlnSmUIWSwDFwlBRZiyPwB-1GhAqYAAvAA+FZrDBLQ4ws4XL5zfCiJg4cw3O5YLI06Gwq4cmSMhwKUhMVitMDyOCi8UAHlpZMZxHwrHUs1pjNEBgUCAAklAAEpYACOlDy4gw3LMq3WwL4oJC4P2JAAgsw2AANAASBrUGm0ACEVVhZUcKQ99PgMPgHOFZqGybj1JotGY3SxWN7fcntAnieTJSKxSH056fQnFfYwPgsQZZhHEpRxYRhPhcm0rAsADRVxwq1gIZwYDvWKt+RmeYMAVQNABlZuVu1W8e96cJ1by14yDFgoxJWLMAPKeABWuFE29EogcLqKYlmooxGzAwGAYAAsmLWNYwFqSKIlAcXZfx3JJnBIWBEDcCgSBgJhPGwKAqwMK8HFYW8LlEB8mCfGwXzfT98h-P9f0A4CSOSEJIKQYQYLAOCEKwJDElIRg-XEMRZwuLQtVmdwvB8KtWIYI0MCYYgMgPMwAFYe0SBwiDbKSwGPM8MiFRTSFYS4JCYfA4IALwkWYsgUwglLMVTz3zClF1sKthTgLRiHyEzHOcqYAFFlkwsxvMwmzKXWMAAB8yBwrANgchRlFmKVlD9FMlTyJxhHFI4yNmehCHMpRPEoKAoAkDgwtoTx4E8EqwFoDA4BwDxMKq2gT1VYQmvEBhRCa1Jd1IWhNSYXIbiwsx+O8IEq0sGJWCgZxSCNU1zVmF0cvwJRZVlKdE2jUR8DMI5AqnEKVtyjatqyHa9rAA7SQpLbK0ecy0tmwh5pSlISjTVb1s27bmSum6Cy2xlln09ZmSwS5dtEJsTIsZkm1GjxxsCzw4DgGgYjk-QECYLUVGIbAxD0mBlLRjHd2ERkGFYSQ1DOJgsAAOTbT65k7ezEhpyQvV3bBCBZ3IF3mTmwC6e0BBCACgLAAASUtMx9RLA2DBNSTABWs2VrQgzlMMGh6R09kGV13UV7N-S0NX0Q4jBWA1s2tZzHXVbDeE+H+2YjkZXItQoYWOcZW4YBHBY7UN-8yLlzWledtWSQdjMncthMDYdEgBHyROywNNWGzAy200d2OU-1qtLv27c+f3SzT3PRl4Zh1gkYEibEkbpsjht0OtGx3hiEWiwzBQ69ZZUODMIWs0LFCsAPXfWcvVQqfzT0cWI9ImX5eL3O3YT7ek-LMuQUljOCCzhXdT2GRY2cXUidEMAbYuO2wChjmuHX9OknP+2FZXiwV9wg3xRHnKsTYuS9nMOkGQ+QvQxAwICMwxQsguVSuKIuh8LYpjzLdQs0piyyhjtg3MIAyQUjCsQ7WsoyF930BIBQhB4HCEQRIZBaIcrODMI+SKgUeEYDoUsS+18sC30IPfTCjIsBqVEFkdYvls6sCARIEBd8H5LBtFFfwWxT4-37Ioo0rB0FYGUZyMRYDGziiyFA1gMCpjMNYYQdhaCPqYJzrggs-C6G8AYc4BxSCwAoN8U48KT4+ERQEVWIRZtTGqPEQ-KRMi5EPzcUokRYiJFiA0VScO39pamwzFKGUIY974OKUQneFY066Mzn-M27tkhpSRBkZwCYkyW2WjvbWusSlkmybaKsbTGkplScnHBYZomFKLHrB6+g2nBx7l2AuKZOlYOoRM1JRTCHH30APaeoghnO1GSXcZfTNnTN6YybANBxDzMICHdm1glnDO0KsnO3TXZnMUVsmZjIigHLDFkBZjzuzLNecckhLsZmTLYD8y5yE+Z3IeXZZ5zs3nmw+dC85BDfkOWMEixZHtdrYSfKijpELMW9JhawOFqcHJYgJSC3uRL8AksimSlZFLnY9LzNi8pOy+yiBwAYRlKKWVsoEWCrQ6Kxkqyxd8i5dLEhOjic3d2-cTT7NSQA0QsTRGgP4WSbxYAXFpQlP-d6Zq9XmMNXQ8ca90Anx2Hoi+ZtPKcMINbTqL97busYQ0o5ijZVWwFSay1GCg1H1mQibAAcOgOp0c6-J0c3Uevjoov1rSy6Jt6LUjNDAnAuFCSIOpGZ3YKzMPkuAUBFGMlpNUX+8poTRqhACPkDb+xNrbWuFtYpaRZFBjAM0ZgYhKCWOTTGVNlzKnEO0zlkbIU8tHV8hWjJ9IPKrTWuFAA6ddDdhR81xOuycOBThmAAGKUFiCydQSwshEBCSdNaspl2BUvdelgwgHVOt6OQKgNBt3okLaINVCtPIFucBeVAAQgA)
But it sure doesn't look right.

**Comment by bassettsj** [comment]
@marudor Still seems like the best solution considering! I tried a few hacky things like that to solve it before, but I couldn't get it did not work. I will open a PR for that and leave a comment about why changing it from interface to class worked somehow.

I am not sure if there was a regression in flow 0.37.x or not.

**Comment by marudor** [comment]
fixed with d28912a0365e24da7e7a99d7b6a787c1e551a2f1
